src/rgw/rgw_user.h

```diff
@@ -221,6 +222,9 @@ struct RGWUserAdminOpState {
   RGWQuotaInfo bucket_quota;
   RGWQuotaInfo user_quota;
 
+  uint32_t user_type;
+  bool user_type_specified = false;
```

Comment:
Would you consider `boost::optional` for `user_type`? It allows to avoid the extra `user_type_specified` member.